### PR TITLE
add no console rule front-end

### LINKED ISSUE #10360: Improve linting rules for frontend codebase in regards to console statements
**Is your feature request related to a problem? Please describe.**

For context, see https://github.com/forem/forem/pull/10136#discussion_r489612418 . Basically there is almost never a scenario where we want console statements in the codebase, so the proposal is to make them lint errors instead of warnings. If there is a valid use case, they could be disabled on a case per case basis.

**Describe the solution you'd like**

Modify eslint configuration files to modify the behavior of the [no-console](https://eslint.org/docs/rules/no-console) rule so that it is registered as an error.

**Describe alternatives you've considered**

N/A

**Additional context**

N/A

### PR DESCRIPTION
**Pull request by narender2031**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
add front-end rule for no console. 

## Related Tickets & Documents
Closes #10360 add no-console rule 

## QA Instructions, Screenshots, Recordings

_Please replace this line with instructions on how to test your changes, as well
as any relevant images for UI changes._

## Added tests?

- [ ] yes
- [x] no, because they aren't needed
- [ ] no, because I need help

## Added to documentation?

- [ ] docs.forem.com
- [ ] readme
- [x] no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

**Review by rhymes — COMMENTED** [review]
Thanks @narender2031!

I noticed we have 3 eslintrc files, should we add this in the other ones as well?

@nickytonline should we also cleanup all the `console.log` and `console.error` around the code?

**Comment by nickytonline** [comment]
> Thanks @narender2031!
> 
> I noticed we have 3 eslintrc files, should we add this in the other ones as well?
> 
> @nickytonline should we also cleanup all the `console.log` and `console.error` around the code?

@rhymes and @narender2031, it can probably just go in the .eslintrc.js that is in the root folder. It should get inherited in the other eslint configurations.

File: app/assets/javascripts/initializers/initializeSwipeGestures.js [hunk 1]
```diff
@@ -63,6 +63,7 @@ function initializeSwipeGestures() {
     // window.onload=function(){
     slideContent(document);
     var h = function (e) {
+      // eslint-disable-next-line no-console
       console.log(e.type, e);
```
Comment (nickytonline) [inline app/assets/javascripts/initializers/initializeSwipeGestures.js:67]: I would just remove this.

**Review by citizen428 — APPROVED** [review]

**Review by Zhao-Andy — APPROVED** [review]
LGTM

**Pull request merged**